pkg/transformers/image.go

```diff
@@ -58,9 +60,9 @@ func (pt *imageTransformer) Transform(resources resmap.ResMap) error {
  finds matched ones and update the image name and tag name
 */
 func (pt *imageTransformer) findAndReplaceImage(obj map[string]interface{}) error {
-	paths := []string{"containers", "initContainers"}
 	found := false
-	for _, path := range paths {
+	for _, fs := range pt.fieldSpecs {
+		path := fs.Path
```

Comment:
Since `containers` and `initContainers` are both corresponding to a list, the code in `updateContainers` only handles this types. You also need to consider other cases since the path from a custom configurations may be corresponding to single container rather than a list.

Reply:
Thanks for the heads up.